modules/mod_comment/models/m_comment.erl

```diff
@@ -78,19 +78,20 @@ m_value(#m{value=undefined}, _Context) ->
 
 
 %% @doc List all comments of the resource.
-%% @spec list_rsc(int(), Context) -> [ PropList ]
-list_rsc(RscId, Context) when is_integer(RscId) ->
+-spec list_rsc(m_rsc:resource(), #context{}) -> list().
+list_rsc(RscId, Context) ->
     F = fun() ->
-        z_db:assoc_props("select * from comment where rsc_id = $1 order by created asc", [RscId], Context)
+        z_db:assoc_props("select * from comment where rsc_id = $1 order by created asc", [m_rsc:rid(RscId, Context)], Context)
```

Comment:
I use `m_rsc:rid/2` in most places to retain the old behaviour when an non-existing resource id integer was passed. 

Alternatively, we may want to use `m_rsc:name_to_id_check/2`, which will crash early if the resource does not exist. Which do you prefer?